app/components/autocomplete/date_suggestion/index.js

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {bindActionCreators} from 'redux';
+import {connect} from 'react-redux';
+
+import {getCurrentTeamId} from 'mattermost-redux/selectors/entities/teams';
+
+import {
+    getMatchTermForDateMention,
+} from 'app/selectors/autocomplete';
+import {getTheme} from 'mattermost-redux/selectors/entities/preferences';
+
+import DateSuggestion from './date_suggestion';
+
+function mapStateToProps(state, ownProps) {
+    const {cursorPosition} = ownProps;
+
+    const value = ownProps.value.substring(0, cursorPosition);
```

Comment:
here you are accessing value through `ownProps` but you are deconstructing in 2 lines above, so maybe you could do `const {cursorPosition, value} = ownProps and here just do `const newValue = value.subtring(0, cursorPosition)` and then `getMatchTermForDateMention(newValue)`